plugins/ida/ida_main.ml

```diff
@@ -22,63 +28,110 @@ module type Target = sig
   end
 end
 
-let get_symbols =
-  Command.create
-    `python
-    ~script:"
-from bap.utils.ida import dump_symbol_info
-dump_symbol_info('$output')
-idc.Exit(0)"
-    ~parser:(fun name ->
-        let blk_of_sexp x = [%of_sexp:string*int64*int64] x in
-        In_channel.with_file name ~f:(fun ch ->
-            Sexp.input_sexps ch |> List.map ~f:blk_of_sexp))
-
-let extract path arch =
-  let id =
-    Data.Cache.digest ~namespace:"ida" "%s" (Digest.file path) in
+let addr_of_mem mem =
+  Memory.min_addr mem
+  |> Bitvector.to_int64
+  |> function
+  | Ok addr -> Some addr
+  | Error _ -> None
+
+let handle_normal_flow =
+  let (!) = Word.of_int64 ~width:32 in
+  function
+  | Some fall -> [Some !fall, `Fall]
+  | None -> []
+
+(** We have to figure out what the kind is of the other jumps. We use
+    the default Bil brancher: if it has a destination that matches one of
+    ours, then we just use that edge type. If not, we just use jump *)
+let handle_other_flow default rest =
+  let (!) = Word.of_int64 ~width:32 in
+  match rest with
+  | [] -> []
+  | l -> List.fold l ~init:[] ~f:(fun acc dest_addr ->
+      match List.find default ~f:(fun (addr,_) -> addr = Some !dest_addr) with
+      (* If IDA dest addr is in default BIL brancher, use same edge info *)
+      | Some (Some addr,kind) -> (Some !dest_addr, kind)::acc
+      (* XXX heuristic: we are assuming `Jump always (when it could be cond) *)
+      | _ -> (Some !dest_addr, `Jump)::acc)
```

Comment:
So it's possible that this jump may under certain conditions be `Cond and not `Jump. I.e., conditional calls in ARM. I use the BIL brancher to get the kind of jump when available. When not, this is just assuming `Jump (there isn't a nice utility function like `val kind_of_jmp : bil -> kind` in the default brancher code... I tried to extract such a utility from the default brancher code, but didn't manage.